lib/cluster/election.c

```diff
@@ -311,8 +312,9 @@ election_vote(election_t *e)
 
     election_reset(e);
     e->state = election_in_progress;
-    vote = pcmk__new_request(e->server, CRM_SYSTEM_CRMD, NULL,
-                             CRM_SYSTEM_CRMD, CRM_OP_VOTE, NULL);
+    message_type = pcmk__server_message_type(e->server);
+    vote = pcmk__new_request(e->server, message_type, NULL,
```

Comment:
It's a bit surprising, nomenclature-wise, that here we use the message type rather than the system type for the `sender_system` and `recipient_system` arguments. It's fine since those arguments are documented as arbitrary strings that may have meaning in the particular message-passing context.

Reply:
Yeah, it's purely for backward compatibility. I would love to get to the point where the primary system name (pacemaker-Xd) is used as the sole IPC endpoint name and message type, but that would involve a lot of hackery for compatibility for a long time.

Reply:
Part of the issue is that we haven't converted all the daemons to use `pcmk_ipc_api_t` yet, which means we still allow external code to use `crm_ipc_new()` with arbitrary endpoint names. That project has to be finished and `crm_ipc_new()` deprecated and dropped before we could drop the old IPC names. If it weren't for that, we could change the IPC names now and map the old names to the new ones in the remote proxy. We could serve both IPC names (no significant technical concerns that) but it wouldn't be worth the ugly.

Reply:
And then of course there's supporting newer remote nodes proxied to older cluster nodes. The tools would have to connect using the old names until we no longer supported cluster nodes without the new names.

Message types are slightly easier. We'd have to add both the old and new names until we stopped supporting rolling upgrades from older versions.